applications/array-optimization.ipynb

```diff
@@ -0,0 +1,240 @@
+    "\n",
+    "Let's actually write the data, and monitor the dashboard while writing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%time _ = inputs_rechunked.store(dest, lock=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this case we end up with many small tasks. Depending on various things (the ratio of machines to cores on the cluster, the memory available on the cluster, bandwidth between machines, network latency), tasks from a single chain may end up on different machines, requiring transfer (highlighted in red). See https://github.com/dask/dask/issues/5105 for an example.\n",
+    "\n",
+    "We can acheive the desired scheduling by having Dask *fuse* chains of tasks. See :ref:`dask.optimize.fuse` for more, but the default values aren't aggressive enough for this computation. We want to ensure that the 25 original inputs (we used `da.random.random`) are fused into a single task. This ensures they'll be executed on a single machine, so the following rechunking will happen on the same machine."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with dask.config.set(fuse_ave_width=25):\n",
+    "    display(inputs_rechunked.blocks[0, :2].visualize(optimize_graph=True))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's perform the write, and again observe the dashboard."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "with dask.config.set(fuse_ave_width=25):\n",
+    "    write = inputs_rechunked.store(dest, lock=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Overall, we notice larger chunks, and no communcation. Each of these contributes to a faster runtime.\n",
+    "\n",
+    "Fusing tasks is not an unambiguously good thing. As the docs for :ref:`dask.optimize.fuse` state\n",
+    "\n",
+    "> This trades parallelism opportunities for faster scheduling by making tasks less granular.\n",
+    "\n",
+    "In this case, *we* know that we can already acheive ideal parallelism by the coarse-grained embarassinly parallel nature of the problem. We're happy to have less fine-grained parllelism since we know we'll still be able to saturate the cluster."
```

Comment:
```suggestion
    "In this case, *we* know that we can already achieve ideal parallelism by the coarse-grained embarrassingly parallel nature of the problem. We're happy to have less fine-grained parallelism since we know we'll still be able to saturate the cluster."
```